biz.aQute.bnd.remote.junit/src/aQute/bnd/remote/junit/JUnitFramework.java

```diff
@@ -311,45 +328,54 @@ public void report() throws InvalidSyntaxException {
 
 	/**
 	 * Show the installed bundles
+	 * 
+	 * @param out the output destination
+	 * @throws IllegalArgumentException if specified argument is {@code null}
 	 */
 	public void reportBundles(Appendable out) {
+		checkNonNullArgument(out, "Output destination instance cannot be null");
 		try (Formatter f = new Formatter(out)) {
 			Stream.of(framework.getBundleContext()
 				.getBundles())
-				.forEach(bb -> {
-					f.format("%4s %s\n", bundleStateToString(bb.getState()), bb);
-				});
+				.forEach(bb -> f.format("%4s %s%n", bundleStateToString(bb.getState()), bb));
 		}
 	}
 
 	/**
 	 * Show the registered service
+	 * 
+	 * @param out the output destination
+	 * @throws IllegalArgumentException if specified argument is {@code null}
 	 */
 	public void reportServices(Appendable out) throws InvalidSyntaxException {
+		checkNonNullArgument(out, "Output destination instance cannot be null");
 		try (Formatter f = new Formatter(out)) {
 			Stream.of(framework.getBundleContext()
-				.getAllServiceReferences(null, null))
-				.forEach(sref -> {
-					f.format("%s\n", sref);
-				});
+				.getServiceReferences((String) null, null))
```

Comment:
This is a semantic change not just a change for an older API. getAllServiceReferences does not care if the service type is not assignable to the caller's bundle. This change can return fewer services and thus may give incorrect results. getAllServiceReferences was added in R4.0, so this method should be used.

Reply:
ah you are right. will change it.